### PR TITLE
parse.d: remove errors.d dependence and some globals

### PR DESCRIPTION
**Pull request by WalterBright**
Removes unnecessary ` import errors;` and reduces use of `global`.

### CODE REVIEW

**Comment by WalterBright** [comment]
More random std.random failures:
```
../dmd/generated/osx/release/64/dmd  -conf= -I../dmd/druntime/import  -w -de -preview=dip1000 -preview=dtorfields -preview=fieldwise -m64 -fPIC -O -release -unittest -version=StdUnittest -c -ofgenerated/osx/release/64/unittest/std/regex/internal/tests.o std/regex/internal/tests.d
****** FAIL debug64 std.random
object.Exception@std/random.d(2145): std.random.uniform(): invalid bounding interval (0, 1)
```
macOS 13.x x64 (M1), DMD (latest)

File: compiler/test/compilable/transition_in.d [hunk 1]
```diff
@@ -5,7 +5,7 @@ TEST_OUTPUT:
 ---
 compilable/transition_in.d(3): Usage of 'in' on parameter
 compilable/transition_in.d(3): Usage of 'in' on parameter
-compilable/transition_in.d(8): Usage of 'in' on parameter
+compilable/transition_in.d(13): Usage of 'in' on parameter
 compilable/transition_in.d(13): Usage of 'in' on parameter
```
Comment (ibuclaw) [inline compiler/test/compilable/transition_in.d:9]: Somethings going wrong here for you to end up with a double error.

**Review by ibuclaw — DISMISSED** [review]
There's something wrong with the error logic, line shouldn't be diagnosed twice.

File: compiler/test/compilable/transition_in.d (hunk 1 above)
Comment (WalterBright) [reply compiler/test/compilable/transition_in.d:9]: It's probably running semantic more than once on a tree, which happens in the compiler. Not worth dealing with it in this PR.

**Comment by dkorpel** [comment]
This is not a refactoring when it changes the behavior of `-transition=in` so it doesn't report usage of `in` inside inactive `version` blocks anymore. @Geod24

**Comment by Geod24** [comment]
True, it changes `-preview=in` and `-transition=in` behavior.

**Comment by WalterBright** [comment]
Which is an improvement, as one would probably like the `version(none)` to not complain about it.

File: compiler/test/compilable/transition_in.d (hunk 1 above)
Comment (WalterBright) [reply compiler/test/compilable/transition_in.d:9]: After all, this PR does not cause semantic to be run twice, that was already happening.

**Comment by dkorpel** [comment]
> Which is an improvement, as one would probably like the `version(none)` to not complain about it.

When you have `version(Windows)` and `version(Posix)` blocks, you do want `-transition=in` to list usages in all of them.

**Comment by WalterBright** [comment]
> When you have version(Windows) and version(Posix) blocks, you do want -transition=in to list usages in all of them.

What if you want to compile with an older compiler? We do some of that right now, in the errors.d source code, to be compatible with the bootstrap compiler.

**Comment by dkorpel** [comment]
> What if you want to compile with an older compiler?

Then you do that? I don't understand what you're getting at.

**Comment by WalterBright** [comment]
```
static if (__VERSION__ < 2092)
    private extern (C++) void noop(const ref Loc loc, const(char)* format, ...) {}
else
    pragma(printf) private extern (C++) void noop(const ref Loc loc, const(char)* format, ...) {}
```
The pragma is not supported in the older compiler.

**Comment by dkorpel** [comment]
What does that have to do with this PR removing some of the messages generated by `-transiton=in`?

**Comment by WalterBright** [comment]
I.e. if the syntax is not permitted in the parser, there is no way to version the code that has to work with older compilers. (Unless you're doing klunky string mixins.)

**Comment by WalterBright** [comment]
> What does that have to do with this PR removing some of the messages generated by -transiton=in?

They may want to conditionally use `in` in the more modern compiler and not be annoyed by messages.

**Comment by dkorpel** [comment]
`-transition=in` is a utility that lists usages of `in` parameters as informational messages, it does not make the compiler error out.

**Comment by WalterBright** [comment]
Sigh. False positive messages are a nuisance.

**Comment by dkorpel** [comment]
False positive in what sense? It's not a warning, it's just a way to locate in parameters. grepping for "in" will lead to many false positives, such as "in" in strings/comments or the `in` operator.

**Review by dkorpel — APPROVED** [review]
I don't care much about `-transition=in`, so this is fine by me

**Comment by ibuclaw** [comment]
> After all, this PR does not cause semantic to be run twice, that was already happening.

Then raise a diagnostic regression so that it can be tracked.

**Comment by WalterBright** [comment]
@ibuclaw 

https://issues.dlang.org/show_bug.cgi?id=23764

**Comment by Geod24** [comment]
@WalterBright : They are not false positive. I want `-transition=in` to list *all* usages, even in `version (none)`. There is a way to make the code work with old compiler, use `[scope] const`.

**Comment by WalterBright** [comment]
I'm open to suggestions on maintaining the goal of the parser being separable from the rest of the compiler, while supporting that.

**Pull request merged**